sdk/analytics/src/main/java/com/klaviyo/analytics/OperationQueue.kt

```diff
@@ -0,0 +1,70 @@
+package com.klaviyo.analytics
+
+import com.klaviyo.analytics.model.Profile
+import com.klaviyo.analytics.networking.ApiClient
+import com.klaviyo.core.KlaviyoException
+import com.klaviyo.core.Registry
+import com.klaviyo.core.config.Clock
+
+internal class OperationQueue {
+    /**
+     * Debounce timer for enqueuing profile API calls
+     */
+    private var timer: Clock.Cancellable? = null
+
+    /**
+     * Pending batch of profile updates to be merged into one API call
+     */
+    private var pendingProfile: Profile? = null
+
+    /**
+     * Uses debounce mechanism to merge profile changes
+     * within a short span of time into one API transaction
+     *
+     * @param profile Incoming profile attribute changes
+     */
+    fun debounceProfileUpdate(profile: Profile) {
+        // Log for traceability
+        val operation = pendingProfile?.let { "Merging" } ?: "Starting"
+        Registry.log.info("$operation profile update")
+
+        // Merge new changes into pending transaction
+        pendingProfile = pendingProfile?.merge(profile) ?: profile
+
+        // Add current identifiers from UserInfo to pending transaction
+        pendingProfile = UserInfo.getAsProfile().merge(pendingProfile!!)
```

Comment:
Is `!!` safe here?

Reply:
Yes because it gets set above, but it could probably be written to avoid using `!!` at all